### PR TITLE
Epoch hash verification

### PR DESCRIPTION
**Pull request by diegomrsantos**
Verify the computed epoch hash against the ones in the file

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java [hunk 1]
```diff
@@ -95,66 +97,101 @@ public class BerkeleySubstateAccumulatorHashStore implements BerkeleyAdditionalS
...
 
   private Database substateAccumulatorHashDatabase;
   private Database epochHashDatabase;
+  private Database lastEpochHashVerifiedDatabase;
 
   private byte[] currentSubstateAccumulatorHash;
   private Optional<Long> lastEpochInDbOpt;
   private Optional<Long> lastEpochInFileOpt;
   private Optional<Long> lastStateVersionInDbOpt;
+  private Optional<Long> lastEpochHashVerified;
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java:133]: Perhaps it worth to consider use of `OptionalLong`.

**Review by siy — APPROVED** [review]
LGTM with one minor comment

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java (hunk 1 above)
Comment (diegomrsantos) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java:133]: Oh thanks for the suggestion, I didn't know it. I will use it in the future, but does it offer any big advantage to justify changing it now?

Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java:133]: It eliminates boxing/unboxing and not susceptible to NPE. If highest possible performance is not a requirement, then there is no significant difference between `OptionalLong` and `Optional<Long>`.

**Pull request merged**